.github/workflows/build.yml

```
@@ -0,0 +1,271 @@
+name: Build 
+on:
+  repository_dispatch:
+  pull_request:
+    types: [assigned, opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+
+env:
+  DOCKERHUB_REPOSITORY:  dfedigital/get-into-teaching-api
+  DOMAIN:                london.cloudapps.digital
+  APPLICATION:           Get Into Teaching API Service
+  PAAS_APPLICATION_NAME: get-into-teaching-api
+
+  
+jobs:
+  turnstyle:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Check workflow concurrency
+        uses: softprops/turnstyle@v1
+        with:
+          poll-interval-seconds: 20
+          same-branch-only: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Slack Notification
+        if: failure()
+        uses: rtCamp/action-slack-notify@master
+        env:
+           SLACK_CHANNEL: getintoteaching_tech
+           SLACK_COLOR: '#3278BD'
+           SLACK_ICON: https://github.com/rtCamp.png?size=48
+           SLACK_MESSAGE: Content Delivery has failed. This is due to a workflow concurrency issue
+           SLACK_TITLE: Turnstyle failure ${{ github.workflow }}
+           SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+           SLACK_FOOTER: ${{env.APPLICATION}}
+
+  build:
+    name: Build and push to DockerHub
+    runs-on: ubuntu-latest
+    needs: turnstyle 
+    outputs:
+       image: ${{steps.docker_image.outputs.IMAGE}}
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@master
+
+      - name: Get Short SHA
+        id: vars
+        run: |
+             echo ::set-output name=sha_short::$(echo $GITHUB_SHA | cut -c -7)
+             if [ ${{github.ref}} == 'refs/heads/master' ]
+             then
+             echo ::set-output name=push::true
+             else
+             echo ::set-output name=push::false
+             fi
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+             ${{ runner.os }}-buildx-
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PERSONAL_ACCESS_TOKEN }}
+
+      - name: Build
+        uses: docker/build-push-action@v2
+        with:
+          push: ${{ steps.vars.outputs.push }}
```

```suggestion
          push: ${{ github.ref == 'refs/heads/master' }}
```
would just adding the condition here work?

---

it might, but I think it might also be confusing when reading back on it in 100 years time ..